multiapi-engine/src/main/java/com/sngular/api/generator/plugin/openapi/OpenApiGenerator.java

```diff
@@ -231,19 +254,27 @@ private String processPath(final String fileSpecPackage, final Boolean isModel)
       path = pathList[0].toPath().resolve(convertPackageToTargetPath(fileSpecPackage, isModel));
     } else {
       path = targetFolder.toPath();
-      path.toFile().mkdirs();
+      File dir = path.toFile();
+      boolean created = dir.mkdirs();
+      if (!created && !dir.isDirectory()) {
+        throw new RuntimeException("Failed to create directory: " + dir);
+      }
       path = path.resolve(convertPackageToTargetPath(fileSpecPackage, isModel));
     }
-    if (!path.toFile().isDirectory()) {
-      path.toFile().mkdirs();
+    File dir = path.toFile();
+    if (!dir.isDirectory()) {
+      boolean created = dir.mkdirs();
+      if (!created) {
+        throw new RuntimeException("Failed to create directory: " + dir);
```

Comment:
Directories are created from a path in 2 different ways in the same function, also, the code is repeated multiple times, wouldn't it be better to extract it to a function?